addons/mail/static/src/models/rtc/rtc.js

```diff
@@ -76,14 +75,15 @@ function factory(dependencies) {
              * This is distinct from this._recoverConnection which tries to restores
              * connection that were established but failed or timed out.
              */
-            this._intervalId = browser.setInterval(() => {
-                if (!this.currentRtcSession) {
+            this._intervalId = browser.setInterval(async () => {
+                if (!this.currentRtcSession || !this.channel) {
                     return;
                 }
-                this._pingServer();
-                if (this.channel) {
-                    this._callSessions();
+                await this._pingServer();
+                if (!this.currentRtcSession || !this.channel) {
+                    return;
                 }
+                this._callSessions();
             }, 30000); // 30 seconds
```

Comment:
I'm not sure whether it's a good idea, but from my understanding of the problem, the time window the list of rtc sessions is wrong only affects the people that just joined at the same time, correct?

If that's the case, maybe we could improve the recovering by shortening the time between join and 1st ping to something like 5 seconds?

All in all, the only way to really ensure good UX is to use DB locks. 30 seconds is a bit long, 5 seconds (just for 1st ping) is still long but at least less critical than 30... but there's still a big server load cost in a short time window when many people join at the same time...

At least there's recovery without page reload, that's already a very good improvement relative to before this PR.